tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/grpc/sample/ProtoBuilderTestCase.java

```diff
@@ -0,0 +1,134 @@
+    public void testUnaryServiceWithPrimitiveParams() {
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "grpcservices", "unary_server1.bal");
+        CompileResult result = BCompileUtil.compile(balFilePath.toAbsolutePath().toString());
+        Assert.assertEquals(result.getErrorCount(), 0, "Compilation errors in unary_server1.bal");
+        Assert.assertEquals(result.getAST().getServices().size(), 1, "File should have one service defined.");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().size(), 1, "Service " +
+                "node should have one default annotation");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().get(0).getAnnotationName
+                ().getValue(), "ServiceDescriptor");
+    }
+
+    @Test(description = "Test compiler plugin for unary service with header params.")
+    public void testUnaryServiceWithHeaderParams() {
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "grpcservices", "unary_service3.bal");
+        CompileResult result = BCompileUtil.compile(balFilePath.toAbsolutePath().toString());
+        Assert.assertEquals(result.getErrorCount(), 0, "Compilation errors in unary_server3.bal");
+        Assert.assertEquals(result.getAST().getServices().size(), 1, "File should have one service defined.");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().size(), 1, "Service " +
+                "node should have one default annotation");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().get(0).getAnnotationName
+                ().getValue(), "ServiceDescriptor");
+    }
+
+    @Test(description = "Test compiler plugin for server streaming service.")
+    public void testServerStreamingService() {
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "grpcservices", "server_streaming_service" +
+                ".bal");
+        CompileResult result = BCompileUtil.compile(balFilePath.toAbsolutePath().toString());
+        Assert.assertEquals(result.getErrorCount(), 0, "Compilation errors in server_streaming_service.bal");
+        Assert.assertEquals(result.getAST().getServices().size(), 1, "File should have one service defined.");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().size(), 1, "Service " +
+                "node should have one default annotation");
+        Assert.assertEquals(result.getAST().getServices().get(0).getAnnotationAttachments().get(0).getAnnotationName
+                ().getValue(), "ServiceDescriptor");
+    }
+
+    @Test(description = "Test compiler plugin for client streaming service.")
+    public void testClientStreamingService() {
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "grpcservices", "client_streaming_service" +
+                ".bal");
```

Comment:
Lets not break at middle of the string value, rather break the line at one of the Paths argument string level.